python_modules/libraries/dagster-airflow/dagster_airflow/dagster_job_factory.py

```diff
@@ -79,3 +85,124 @@ def make_dagster_job_from_airflow_dag(
+
+    Returns:
+        Definitions
+    """
+
+    repo = make_dagster_repo_from_airflow_dag_bag(
+        dag_bag,
+        "dagster_airflow_repo",
+        refresh_from_airflow_db,
+        use_airflow_template_context,
+        mock_xcom,
+        use_ephemeral_airflow_db,
+        connections,
+    )
+
+    return Definitions(
+        schedules=repo.schedule_defs,
+        jobs=repo.get_all_jobs(),
+    )
+
+
+def make_dagster_definitions_from_airflow_dags_path(
+    dag_path,
+    safe_mode=True,
+    store_serialized_dags=False,
+    use_airflow_template_context=False,
+    mock_xcom=False,
+    use_ephemeral_airflow_db=True,
+    connections=None,
+):
+    """Construct Dagster definitions corresponding to Airflow DAGs in dag_path.
+
+    ``DagBag.get_dag()`` dependency requires Airflow DB to be initialized.
+
+    Usage:
+        Create ``make_dagster_definitions.py``:
+
+        .. code-block:: python
+
+            from dagster_airflow.dagster_pipeline_factory import make_dagster_definitions_from_airflow_dags_path
+
+            def make_definitions_from_dir():
+                return make_dagster_definitions_from_airflow_dags_path(
+                    '/path/to/dags/'
+                )
+
+        ``dagit -f path/to/make_dagster_definitions.py``
+
+    Args:
+        dag_path (str): Path to directory or file that contains Airflow Dags
+        safe_mode (bool): True to use Airflow's default heuristic to find files that contain DAGs
+            (ie find files that contain both b'DAG' and b'airflow') (default: True)
+        store_serialized_dags (bool): True to read Airflow DAGS from Airflow DB. False to read DAGS
+            from Python files. (default: False)
+        use_airflow_template_context (bool): If True, will call get_template_context() on the
+            Airflow TaskInstance model which requires and modifies the DagRun table. The use_airflow_template_context
+            setting is ignored if use_ephemeral_airflow_db is True.
+            (default: False)
+        mock_xcom (bool): If True, dagster will mock out all calls made to xcom, features that
+            depend on xcom may not work as expected. (default: False)
+        use_ephemeral_airflow_db (bool): If True, dagster will create an ephemeral sqlite airflow
+            database for each run. (default: False)
+        connections (List[Connection]): List of Airflow Connections to be created in the Ephemeral
+            Airflow DB, if use_emphemeral_airflow_db is False this will be ignored.
+
+    Returns:
+        Definitions
+    """
+
+    repo = make_dagster_repo_from_airflow_dags_path(
+        dag_path,
+        "dagster_airflow_repo",
+        safe_mode,
+        store_serialized_dags,
+        use_airflow_template_context,
+        mock_xcom,
+        use_ephemeral_airflow_db,
+        connections,
+    )
+
+    return Definitions(
+        schedules=repo.schedule_defs,
+        jobs=repo.get_all_jobs(),
+    )
```

Comment:
I think this would feel a lot clearer and safer it `make_dagster_repo_from_airflow_dags_path` was actually `make_schedules_and_jobs_from_airflow_dags_path` as the core workhorse function. A few reasons:

* It's very surprising that only schedules and jobs get shuffled along here.
* And then inspecting the code it calls it actually sort of purports to create assets and doesn't. 
* Finally the way this all implemented we construct a repo, then pass to Definitions, which in turn does some processing, then constructs another repository which I think could cause some unexpected bugs long term.

So to summarize I recommend making a raw workhouse function that produces definitions "outside" a repository or a `Definitions` object, and then only pass those to the enclosing container (repository/Definitions) at the highest possible level in the callstack.

Reply:
Fair points, I think keeping the declaration of `Definitions` in user-code will help make things a bit clearer? If theres a way to derive Definitions from a Repo that might also be interesting?